--- a/src/mod_dirlisting.c
+++ b/src/mod_dirlisting.c
@@ -93,6 +93,7 @@
 	const buffer *external_js;
 	const buffer *encoding;
 	const buffer *set_footer;
+	const buffer *top_directory;
 	const struct dirlist_cache *cache;
 } plugin_config;
 
@@ -363,6 +364,9 @@
       case 16:/* dir-listing.sort */
         pconf->sort = (char)cpv->v.u;
         break;
+      case 17:/* dir-listing.top-directory */
+        pconf->top_directory = cpv->v.b;
+        break;
       default:/* should not happen */
         return;
     }
@@ -435,6 +439,9 @@
      ,{ CONST_STR_LEN("dir-listing.sort"),
         T_CONFIG_BOOL,
         T_CONFIG_SCOPE_CONNECTION }
+     ,{ CONST_STR_LEN("dir-listing.top-directory"),
+        T_CONFIG_STRING,
+        T_CONFIG_SCOPE_CONNECTION }
      ,{ NULL, 0,
         T_CONFIG_UNSET,
         T_CONFIG_SCOPE_UNSET }
@@ -513,6 +520,10 @@
                 break;
               case 16:/* dir-listing.sort */
                 break;
+              case 17:/* dir-listing.top-directory */
+                if (buffer_is_blank(cpv->v.b))
+                    cpv->v.b = NULL;
+                break;
               default:/* should not happen */
                 break;
             }
@@ -980,7 +991,9 @@
 		"</thead>\n"
 		"<tbody>\n"
 	));
-	if (!buffer_is_equal_string(&r->uri.path, CONST_STR_LEN("/"))) {
+	if (!buffer_is_equal_string(&r->uri.path, CONST_STR_LEN("/"))
+	    && (NULL == p->conf.top_directory
+	        || !buffer_is_equal_buf(&r->uri.path, p->conf.top_directory))) {
 		buffer_append_string_len(out, CONST_STR_LEN(
 		"<tr class=\"d\">"
 			"<td class=\"n\"><a href=\"../\">..</a>/</td>"
